File: 240422/약수 출력하기/print-divisor.c
#include <stdio.h>
#include <math.h>

int main() {
    int n;
    scanf("%d", &n);

    int sqrt_n = sqrt(n);

    for (int i = 1; i <= sqrt_n; i++) {
        if (n % i == 0) {
            printf("%d ", i);
        }
    }

    for (int i = sqrt_n; i >= 1; i--) {
        if (n % i == 0 && i != n / i) { 
            printf("%d ", n / i);
        }
    }

    return 0;
}

File: 240422/약수 출력하기/print-divisor.kt
import kotlin.math.sqrt

fun main() {
    val n = readLine()!!.toInt()

    val sqrtN = sqrt(n.toDouble()).toInt()

    for (i in 1..sqrtN) {
        if (n % i == 0) {
            print("$i ")
        }
    }

    for (i in sqrtN downTo 1) {
        if (n % i == 0 && i != n / i) { 
            print("${n / i} ")
        }
    }
}